source/geometry_model/box.cc

```diff
@@ -207,8 +207,15 @@ namespace aspect
     double
     Box<dim>::depth(const Point<dim> &position) const
     {
-      // this depth does not take topography into consideration
-      const double d = maximal_depth()-(position(dim-1)-box_origin[dim-1]);
+      // Get the surface x (,y) point
+      Point<dim-1> surface_point;
+      for (unsigned int d=0; d<dim-1; ++d)
+        surface_point[d] = position[d];
+
+      // Get the surface topography at this point
+      const double topo = topo_model->value(surface_point);
+
+      const double d = maximal_depth()-(position(dim-1)-box_origin[dim-1])+topo;
```

Comment:
Are you sure this is correct?` maximal_depth` includes the maximum topography to which you add the local topography as well. Shouldn't it be: 
`d = extents[dim-1] + topo - (position(dim-1)-box_origin[dim-1]);`

Reply:
Good catch! You are correct :) Although I checked the results visually, I didn't notice that the scale didn't start at depth zero :(